src/entity.ts

```diff
@@ -134,7 +134,7 @@ export class Entity {
 		if (!this._context)
 			this._context = context;
 		// Ensure provided context waits on the existing context to be ready
-		else
+		else if (this._context !== context)
 			context.wait(new Promise(resolve => this._context.ready(resolve)));
```

Comment:
This is obviously outside of the scope of these changes, but I wonder if this code might read better with something like `context.wait(this._context.asPromise())`, or `context.waitForContext(this._context)`, or some other variation? Or, `ready` could return a promise so it could be called like `context.wait(this._context.ready())`? Or, the context itself could be promise-like (essentially `ready` replaced with `then`) so that it could be called like `context.wait(this._context)`? Anyway, just throwing the idea out there for consideration since we're looking at this code at the moment.

Reply:
Implemented this idea.